src/utils/breakdown_validation.py

```diff
@@ -0,0 +1,112 @@
+            msg += f"Columns 222 + 223 do not equal column 203 for reference: {row['reference']}.\n "
+        if row['check2']:
+            bool_dict[index] = False
+            msg += f"Columns 202 + 223 do not equal column 204 for reference: {row['reference']}.\n "
+        if row['check3']:
+            bool_dict[index] = False
+            msg += f"Columns 205 + 206 + 207 do not equal column 204 for reference: {row['reference']}.\n "
+        if row['check4']:
+            bool_dict[index] = False
+            msg += f"Columns 221 is greater than column 209 for reference: {row['reference']}.\n "
+        if row['check5']:
+            bool_dict[index] = False
+            msg += f"Columns 219 + 220 + 209 do not equal column 210 for reference: {row['reference']}.\n "
+        if row['check6']:
+            bool_dict[index] = False
+            msg += f"Columns 204 + 210 + 209 do not equal column 211 for reference: {row['reference']}.\n "
+        if row['check7']:
+            bool_dict[index] = False
+            msg += f"Columns 212 + 246 do not equal 218 and Columns 212 + 246 do not equal column 211 for reference: {row['reference']}.\n "
+        if row['check8']:
+            bool_dict[index] = False
+            msg += f"Columns 225 + 226 + 227 + 228 + 229 + 237 do not equal column 218 for reference: {row['reference']}.\n "
+        if row['check9']:
+            bool_dict[index] = False
+            msg += f"Columns 302 + 303 + 304 do not equal column 305 for reference: {row['reference']}.\n "
+        if row['check10']:
+            bool_dict[index] = False
+            msg += f"Columns 501 + 503 + 505 do not equal column 507 for reference: {row['reference']}.\n "
+        if row['check11']:
+            bool_dict[index] = False
+            msg += f"Columns 502 + 504 + 506 do not equal column 508 for reference: {row['reference']}.\n "
+        if row['check12']:
+            bool_dict[index] = False
+            msg += f"Columns 405 + 407 + 409 do not equal column 411 for reference: {row['reference']}.\n "
+        if row['check13']:
+            bool_dict[index] = False
+            msg += f"Columns 406 + 408 + 410 do not equal column 412 for reference: {row['reference']}.\n " 
+        if True in bool_dict.values():
+            bool_dict[index] = True
+            msg += f"All breakdown values are valid for references: {row['reference']}.\n"
+    
+    return bool_dict, msg
+
+def run_breakdown_validation(df: pd.DataFrame) -> pd.DataFrame:
+    """Runs the breakdown_validation function and outputs the msg to the logger"""
+    not_null_df = df.dropna(subset=['222', '223', '203', '202', '204', '205', '206', '207', '221', '209', '219', '220', '210', '204', '211', '212', 
+    '214', '216', '242', '250', '243', '244', '245', '246', '218', '225', '226', '227', '228', '229', '237', '302', '303', '304', '305', '501', '503', 
+    '505', '507', '502', '504', '506', '508', '405', '407', '409', '411', '406', '408', '410', '412'])
```

Comment:
This is looking for a null in any of these columns - so we need to add how = 'all' so we are only excluding the rows that are missing values in all these columns.